pkg/broker/config/key.go

```diff
@@ -30,51 +30,98 @@ import (
 	"knative.dev/pkg/metrics/metricskey"
 )
 
-// BrokerKey uniquely identifies a single Broker, at a given point in time.
-type BrokerKey struct {
-	namespace string
-	name      string
+var (
+	// cellTenantTypeFromLowerCase maps the lower case string to its type.
+	// E.g. 'broker' -> CellTenantType_BROKER. It is filled in by init().
+	cellTenantTypeFromLowerCase map[string]CellTenantType
+	// cellTenantTypeToLowerCase maps the type to its lower case string.
+	// E.g. CellTenantType_BROKER -> 'broker'. It is filled in by init().
+	cellTenantTypeToLowerCase map[CellTenantType]string
+)
+
+func init() {
+	cellTenantTypeFromLowerCase = make(map[string]CellTenantType, len(CellTenantType_value))
+	cellTenantTypeToLowerCase = make(map[CellTenantType]string, len(CellTenantType_value))
+	for k, v := range CellTenantType_value {
+		t := CellTenantType(v)
+		lowerCase := strings.ToLower(k)
+		cellTenantTypeFromLowerCase[lowerCase] = t
+		cellTenantTypeToLowerCase[t] = lowerCase
+	}
+}
```

Comment:
Didn't know we need to implement this ourselves, was thinking about changing the protobuf, but it seems that protobuf does not like lower case enum. I looked at the generated code and we are just adding another enum to string mapping function here, without any performance impact at all. So this is good too.

Reply:
I think the lower case in URLs looks nicer, so I think I'll leave it like this.